File: src/main/java/com/woomulwoomul/woomulwoomulbackend/domain/user/ProviderType.java
package com.woomulwoomul.woomulwoomulbackend.domain.user;

import lombok.Getter;

@Getter
public enum ProviderType {
    KAKAO("카카오"),
    ;

    private final String text;

    ProviderType(String text) {
        this.text = text;
    }
}

File: src/main/kotlin/com/woomulwoomul/woomulwoomulbackend/domain/user/ProviderType.kt
package com.woomulwoomul.woomulwoomulbackend.domain.user

import com.woomulwoomul.woomulwoomulbackend.common.constant.ExceptionCode.PROVIDER_TYPE_INVALID
import com.woomulwoomul.woomulwoomulbackend.common.response.CustomException
import jodd.util.StringUtil

enum class ProviderType {
    KAKAO
    ;

    companion object {
        fun of(providerType: String): ProviderType {
            val type = providerType.uppercase()
            if (StringUtil.equals(type, KAKAO.name))
                return KAKAO
            else
                throw CustomException(PROVIDER_TYPE_INVALID)
        }
    }
}